File: java/JeuDeCarte/src/Paquet.java
import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

public class Paquet {
    final private List<Cartes> cartes;
    private int nbCarte;

    //Constructeur du paquet
    public Paquet() {
        nbCarte = 0;
        cartes = new ArrayList<>();
    }

    //Mélanger les cartes
    public void melanger() {
        Collections.shuffle(this.cartes);
    }

    // Ajouter une carte au paquet
    public void ajoutCarte(Cartes newCarte) {
        cartes.add(newCarte);
        nbCarte++;
    }

    //retirer la première carte
    public Cartes retirerCarte() {
        Cartes carte = cartes.get(0);
        cartes.remove(0);
        nbCarte--;
        return carte;
    }

    //Retirer une carte particulière.
    public Cartes retirerCarte(int index) {
        Cartes carte = cartes.get(index);
        cartes.remove(index);
        nbCarte--;
        return carte;
    }

    //fonctions retournant l'index d'une carte ayant soit la meme couleur ou la meme valeurs que la carte donner
    public int checkCarte(Cartes carte){
        for(int i = 0; i < cartes.size(); i++){
            if (cartes.get(i).getValeur() == carte.getValeur() || cartes.get(i).getCouleur() == carte.getCouleur())
                return i;
        }
        return -1;
    }

    //Retourner le nombre de cartes
    public int getNbCarte() {
        return nbCarte;
    }

    //retourner la main du joueur
    public List<Cartes> getCartes() {
        return cartes;
    }
}

File: c#/JeuDeCarte-C/JeuDeCarte-C/Paquet.cs
namespace CardGame;
// Paquet de cartes, joueur, pioche et dépot
public class Paquet
{
    private List<Cartes> cartes;
    private int nbCarte;

    public Paquet()
    {
        nbCarte = 0;
        cartes = new List<Cartes>();
    }

    //Mélanger un paquet.
    public void Melanger()
    {
        Random rng = new Random();
        int n = cartes.Count;
        while (n > 1)
        {
            n--;
            int k = rng.Next(n + 1);
            Cartes value = cartes[k];
            cartes[k] = cartes[n];
            cartes[n] = value;
        }
    }

    //Ajouter une carte au paquet.
    public void AjoutCarte(Cartes newCarte)
    {
        cartes.Add(newCarte);
        nbCarte++;
    }

    //Retirer la première carte du paquet.
    public Cartes RetirerCarte()
    {
        Cartes carte = cartes[0];
        cartes.RemoveAt(0);
        nbCarte--;
        return carte;
    }
    
    //Retirer une carte précise d'un paquet.
    public Cartes RetirerCarte(int index)
    {
        Cartes carte = cartes[index];
        cartes.RemoveAt(index);
        nbCarte--;
        return carte;
    }

    //Vérifier la valeur d'une carte si elle est jouable.
    public int CheckCarte(Cartes carte)
    {
        for (int i = 0; i < cartes.Count; i++)
        {
            if (cartes[i].GetValeur() == carte.GetValeur() || cartes[i].GetCouleur() == carte.GetCouleur())
            {
                return i;
            }
        }
        return -1;
    }

    //Retourner le nombre de carte d'un paquet.
    public int GetNbCarte()
    {
        return nbCarte;
    }

    //Retourner la liste de cartes.
    public List<Cartes> GetCartes()
    {
        return cartes;
    }
}